plugins/catalog-backend-module-gitlab/src/providers/GitlabDiscoveryEntityProvider.test.ts

```diff
@@ -558,4 +558,125 @@ describe('GitlabDiscoveryEntityProvider - events', () => {
     expect(entityProviderConnection.applyMutation).toHaveBeenCalledTimes(0);
   });
   // EventSupportChange: stop add tests >>>
+
+  // add the following test:
+  // setup: 3 projects, 2 groups
+  // 1 test where none match --> OK
+  // 1 test where match only 1 group --> OK
+  // 1 test where match 2 groups
+  // 1 test where match 1 group with 2 identical regex
+
+  it('should ignore projects when none of the groups regex patterns match', async () => {
+    const config = new ConfigReader(mock.config_groupPatterns_only_noMatch);
+
+    const schedule = new PersistingTaskRunner();
+    const entityProviderConnection: EntityProviderConnection = {
+      applyMutation: jest.fn(),
+      refresh: jest.fn(),
+    };
+    const provider = GitlabDiscoveryEntityProvider.fromConfig(config, {
+      logger,
+      schedule,
+    })[0];
+
+    await provider.connect(entityProviderConnection);
+
+    await provider.refresh(logger);
+
+    expect(entityProviderConnection.applyMutation).toHaveBeenCalledWith({
+      type: 'full',
+      entities: [],
+    });
+
+    expect(entityProviderConnection.applyMutation).toHaveBeenCalledTimes(1); // No entities should be applied
+  });
+});
```

Comment:
You didn't mean to have the closing brace etc here right? It's all wrapped in `describe` blocks typically